includes/class-edd-customer.php

```diff
@@ -396,8 +397,8 @@ public function set_primary_email( $new_primary_email = '' ) {
 		// Remove new primary from list of additional emails
 		$remove = $this->remove_email( $new_primary_email );
 
-		// Add old email to additional emails list
-		$add = $this->add_email( $old_email );
+		// Add old email to additional emails list and if the email was forcibly put there, make sure it is re-added
+		$add = $this->add_email( $old_email, false, ( email_exists( $old_email ) ) );
```

Comment:
The `email_exists( $old_email )` check needs to be pulled out of the `add_email` method in order to main PHP 5.2 compatibility. Otherwise we'll get the fatal error of using a function in a write context.